### PR TITLE
add new CUAV Nora board

### PR DESCRIPTION
**Pull request by bkueng**
Rebased version of  #14272.

@CUAVcaijie can you test this? I added some changes in abe16d7242d686ec11627b8a32eee6699cc1ccad.

TODO:
 - [ ] add to CI
 - [ ] adis16470 driver
 - [x] icm20649 driver

### CODE REVIEW

File: ROMFS/px4fmu_common/init.d/rc.interface [hunk 1]
```diff
@@ -277,6 +277,12 @@ then
 			pwm disarmed -c 6 -p p:PWM_AUX_DIS6 -d ${OUTPUT_AUX_DEV}
 			pwm disarmed -c 7 -p p:PWM_AUX_DIS7 -d ${OUTPUT_AUX_DEV}
 			pwm disarmed -c 8 -p p:PWM_AUX_DIS8 -d ${OUTPUT_AUX_DEV}
+			pwm disarmed -m 0x100 -p p:PWM_MAIN_DIS9 -d ${OUTPUT_AUX_DEV}
```
Comment (dagar) [inline ROMFS/px4fmu_common/init.d/rc.interface]: What's going on here?

Comment (bkueng) [reply ROMFS/px4fmu_common/init.d/rc.interface]: You mean the `-m`? Because `-c 10` would select channels 1 and 0.

**Comment by hamishwillee** [comment]
Hi @bkueng You guys got a plan to add docs for this board?

**Comment by CUAVcaijie** [comment]
> 重新定位的版本 [＃14272](https://github.com/PX4/Firmware/pull/14272)。
> 
> @CUAVcaijie你可以测试吗？我在[abe16d7中](https://github.com/PX4/Firmware/commit/abe16d7242d686ec11627b8a32eee6699cc1ccad)添加了一些更改。
> 
> 去做：
> 
> * [ ]  添加到CI
> * [ ]  adis16470驱动程序
> * [ ]  icm20649驱动程序

Ok

File: ROMFS/px4fmu_common/init.d/rc.interface (hunk 1 above)
Comment (dagar) [reply ROMFS/px4fmu_common/init.d/rc.interface:280]: Ha, seems like a good reminder we should find a new way to do this.

**Comment by CUAVcaijie** [comment]
> 重新定位的版本 [＃14272](https://github.com/PX4/Firmware/pull/14272)。
> 
> @CUAVcaijie你可以测试吗？我在[abe16d7中](https://github.com/PX4/Firmware/commit/abe16d7242d686ec11627b8a32eee6699cc1ccad)添加了一些更改。
> 
> 去做：
> 
> * [ ]  添加到CI
> * [ ]  adis16470驱动程序
> * [ ]  icm20649驱动程序

when I use CUAV Nora to debug ADIS16470, when I use int SPI :: transferhword (uint16_t * send, uint16_t * recv, unsigned len) function, the program is stuck.

**Comment by zhaoxiaowei1013** [comment]
> > 重新定位的版本 [＃14272](https://github.com/PX4/Firmware/pull/14272)。
> > @CUAVcaijie你可以测试吗？我在[abe16d7中](https://github.com/PX4/Firmware/commit/abe16d7242d686ec11627b8a32eee6699cc1ccad)添加了一些更改。
> > 去做：
> > 
> > * [ ]   添加到CI
> > * [ ]   adis16470驱动程序
> > * [ ]   icm20649驱动程序
> 
> when I use CUAV Nora to debug ADIS16470, when I use int SPI :: transferhword (uint16_t * send, uint16_t * recv, unsigned len) function, the program is stuck.

hard fault?

**Comment by CUAVcaijie** [comment]
> Rebased version of #14272.
> 
> @CUAVcaijie can you test this? I added some changes in [abe16d7](https://github.com/PX4/Firmware/commit/abe16d7242d686ec11627b8a32eee6699cc1ccad).
> 
> TODO:
> 
> * [ ]  add to CI
> * [ ]  adis16470 driver
> * [ ]  icm20649 driver

when I use CUAV Nora to debug ADIS16470, when I use int SPI :: transferhword (uint16_t * send, uint16_t * recv, unsigned len) function, the program is stuck.

**Comment by CUAVcaijie** [comment]
> > > 重新定位的版本 [＃14272](https://github.com/PX4/Firmware/pull/14272)。
> > > @CUAVcaijie你可以测试吗？我在[abe16d7中](https://github.com/PX4/Firmware/commit/abe16d7242d686ec11627b8a32eee6699cc1ccad)添加了一些更改。
> > > 去做：
> > > 
> > > * [ ]    添加到CI
> > > * [ ]    adis16470驱动程序
> > > * [ ]    icm20649驱动程序
> > 
> > 
> > when I use CUAV Nora to debug ADIS16470, when I use int SPI :: transferhword (uint16_t * send, uint16_t * recv, unsigned len) function, the program is stuck.
> 
> hard fault?

It is possible that the H7 library does not yet support 16-bit SPI transmission

**Comment by thopiekar** [comment]
The board has been added already. Can this be closed?

**Comment by dagar** [comment]
> The board has been added already. Can this be closed?

Yes, although one missing piece in master is still supporting all 14 PWM. Let's bring that in separately.

**Pull request closed without merging**